core/src/test/java/io/confluent/rest/CustomMethodsTest.java

```diff
@@ -0,0 +1,165 @@
+import javax.ws.rs.core.Configurable;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.eclipse.jetty.server.Server;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Purpose : To test rest-utils / java.ws.rs code's ability to support
+ *  "custom methods" that use a colon.
+ * I.e. can we have two endpoints like
+ *  - /foo/v2/collection/{id}
+ *  - /foo/v2/collection/{id}:myverb
+ */
+public class CustomMethodsTest {
+  private TestRestConfig config;
+  private Server server;
+
+  @Before
+  public void setUp() throws Exception {
+    Properties props = new Properties();
+    props.setProperty("listeners", "http://localhost:0");
+    config = new TestRestConfig(props);
+    CustomMethodsApplication application = new CustomMethodsApplication(config);
+    server = application.createServer();
+    server.start();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    server.stop();
+    server.join();
+  }
+  private static class CustomMethodsApplication extends Application<TestRestConfig> {
+
+    CustomMethodsApplication(TestRestConfig props) {
+      super(props);
+    }
+
+    @Override
+    public void setupResources(Configurable<?> config, TestRestConfig appConfig) {
+      config.register(CustomMethodsResource.class);
+    }
+  }
+
+  @Produces(MediaType.APPLICATION_OCTET_STREAM)
+  @Path("/foo")
+  public static class CustomMethodsResource {
+
+    @GET
+    @Path("/v2/collection/{id}")
+    public String baseEndpoint(@PathParam("id") final String id) {
+      return "base " + id;
+    }
+
+    @GET
+    @Path("/v2/collection/{id}:search")
+    public String customMethodEndpoint(@PathParam("id") final String id) {
+      return "custom " + id;
+    }
+  }
+
+  @Test
+  public void testBaseCall() {
+    Response response = ClientBuilder.newClient()
+        .target(server.getURI())
+        .path("/foo/v2/collection/pants")
+        .request().get();
+    assertEquals(200, response.getStatus());
+    String responseBody = response.readEntity(String.class);
+    assertEquals("base pants", responseBody);
+  }
+
+  @Test
+  public void testIncorrectBaseCall() {
+    // namely that you if you have an id of `pants:search` then it will
+    // find the custom-method
+    Response response = ClientBuilder.newClient()
+            .target(server.getURI())
+            .path("/foo/v2/collection/pants:search")
+            .request().get();
+    assertEquals(200, response.getStatus());
+    String responseBody = response.readEntity(String.class);
+    assertEquals("custom pants", responseBody);
+  }
+
+
+  @Test
+  public void testCustomMethodCall() {
+
+    String[] tests = new String[]{
+            "pants",
+            "shoes",
+            "pants-shoes", // test hyphen
+            "pants:shoes", // test colon
+            "pants:search", // test colon + string that matches custom method
+            "pants:search:search", // why not
```

Comment:
cool that the custom method works even with colons in the IDs too.

but of course, we can't have colons in the IDs because the base method doesn't work (as you noted above)